zerver/lib/bugdown/fenced_code.py

```diff
@@ -111,9 +111,20 @@
 
 class FencedCodeExtension(markdown.Extension):
 
+    def __init__(self, *args, **kwargs):
+        # type: (*Any, **Union[bool, None, Text]) -> None
+        self.config = {
+            "code_block_processor_disabled": [kwargs["code_block_processor_disabled"], "Disabled for email gateway"],
+        }
+        super().__init__(*args, **kwargs)
+
     def extendMarkdown(self, md, md_globals):
         # type: (markdown.Markdown, Dict[str, Any]) -> None
         """ Add FencedBlockPreprocessor to the Markdown instance. """
+        # Disable code block processor for email gateway.
+        if self.getConfig("code_block_processor_disabled"):
+            del md.parser.blockprocessors["code"]
+
```

Comment:
I think it would be better to put this into `__init__.py`, because (1) this is the processor for non-fenced code blocks, and (2) `fenced_code.py` is third-party code that we don't want to grow too different from upstream.

Reply:
I think things like adding/removing processors can be done only in `extendMarkdown()`?

Reply:
Yeah, so do it in the `extendMarkdown()` for `class Bugdown`, not for `fenced_code`.